src/client/canvas/object/shape.ts

```diff
@@ -0,0 +1,18 @@
+import { observable } from 'mobx'
+import { Appearance } from '../appearance/appearance'
+import { BasicAppearance } from '../appearance/basic_appearance'
+import { Geometry } from '../geometry/geometry'
+
+export class Shape<T> {
```

Comment:
Nope, the renderer would then need to be cast everywhere as it can deduce exactly which properties are available for each shape. I'd need to refactor the renderer to be more explicit if I were to do this. This can also be addressed in the next PR I do.